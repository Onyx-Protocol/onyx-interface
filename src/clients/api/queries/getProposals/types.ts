import { Proposal } from 'types';

export interface GetProposalsInput {
  limit?: number;
  page?: number;
}

export interface GetProposalInput {
  id: number | string;
}

export interface ProposalApiResponse {
  call_datas?: string | null;
  callDatas?: string | null;
  signatures: string;
  targets: string;
  values: string;
  against_votes?: string | null;
  againstVotes?: string | null;
  cancel_block?: number | null;
  cancel_timestamp?: number | null;
  cancel_tx_hash?: string | null;
  cancelBlock?: number | null;
  cancelTimestamp?: number | null;
  cancelTxHash?: string | null;
  canceled: number | boolean;
  created_at?: string | null;
  createdAt?: string | null;
  created_block: number | null;
  created_timestamp: number | null;
  created_tx_hash?: string | null;
  description: string;
  end_block?: number;
  end_timestamp?: number | null;
  end_tx_hash?: string | null;
  endBlock?: number;
  endTimestamp?: number | null;
  endTxHash?: string | null;
  eta: number;
  executed: number | boolean;
  executed_block: number | null;
  executed_timestamp: number | null;
  executed_tx_hash: string | null;
  for_votes?: string | null;
  forVotes?: string | null;
  id: number;
  proposer: string;
  queued_block: number | null;
  queued_timestamp: number | null;
  queued_tx_hash: string | null;
  start_block?: number | null;
  start_timestamp?: number;
  start_tx_hash?: string | null;
  startBlock?: number | null;
  startTimestamp?: number;
  startTxHash?: string | null;
  state:
    | 'Pending'
    | 'Active'
    | 'Canceled'
    | 'Defeated'
    | 'Successded'
    | 'Queued'
    | 'Expired'
    | 'Executed';
  // JSON
  updated_at?: string | null;
  updatedAt?: string | null;
  voter_count?: number | null;
  voterCount?: number | null;
}

export interface ProposalsApiResponse {
  data: ProposalApiResponse[];
  metadata: {
    page: number;
    limit: number;
    totalItem: number;
    totalPage: number;
  };
}

export interface ProposalDetailApiResponse {
  data: ProposalApiResponse;
  metadata: {
    page: number;
  };
}

export interface GetProposalsOutput {
  page: number;
  limit: number;
  total: number;
  proposals: Proposal[];
}

export type GetProposalOutput = Proposal;
