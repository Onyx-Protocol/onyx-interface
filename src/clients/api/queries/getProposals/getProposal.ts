import { VError } from 'errors';
import { formatToProposal, restService } from 'utilities';

import { GetProposalInput, GetProposalOutput, ProposalDetailApiResponse } from './types';

const getProposal = async ({ id }: GetProposalInput): Promise<GetProposalOutput> => {
  const response = await restService<ProposalDetailApiResponse>({
    endpoint: `/proposal/${id}`,
    method: 'GET',
    gov: true,
  });

  const payload = response.data?.data;

  // @todo Add specific api error handling
  // if ('result' in response && response.result === 'error') {
  //   throw new VError({
  //     type: 'unexpected',
  //     code: 'somethingWentWrong',
  //     data: { message: response.message },
  //   });
  // }

  if (!payload) {
    throw new VError({ type: 'unexpected', code: 'somethingWentWrong' });
  }

  const convertedPayload = {
    ...payload.data,
    for_votes: payload.data.forVotes,
    against_votes: payload.data.againstVotes,
    start_block: payload.data.startBlock,
    start_tx_hash: payload.data.startTxHash,
    start_timestamp: payload.data.startTimestamp,
    cancel_block: payload.data.cancelBlock,
    cancel_tx_hash: payload.data.cancelTxHash,
    cancel_timestamp: payload.data.cancelTimestamp,
    end_block: payload.data.endBlock,
    end_tx_hash: payload.data.endTxHash,
    end_timestamp: payload.data.endTimestamp,
    voter_count: payload.data.voterCount,
    created_at: payload.data.createdAt,
    updated_at: payload.data.updatedAt,
  };

  return formatToProposal(convertedPayload);
};

export default getProposal;
