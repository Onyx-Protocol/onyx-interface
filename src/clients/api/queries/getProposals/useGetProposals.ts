import { QueryObserverOptions, useQuery } from 'react-query';

import { GetProposalsInput, GetProposalsOutput, getProposals } from 'clients/api';
import FunctionKey from 'constants/functionKey';

type Options = QueryObserverOptions<
  GetProposalsOutput,
  Error,
  GetProposalsOutput,
  GetProposalsOutput,
  [FunctionKey.GET_PROPOSALS, GetProposalsInput]
>;

const useGetProposals = (params: GetProposalsInput = {}, options?: Options) =>
  // This endpoint is paginated so we keep the previous responses by default to
  // create a more seamless paginating experience
  useQuery([FunctionKey.GET_PROPOSALS, params], () => getProposals(params), {
    keepPreviousData: true,
    placeholderData: { proposals: [], total: 0 },
    refetchInterval: undefined,
    ...options,
  });

export default useGetProposals;
