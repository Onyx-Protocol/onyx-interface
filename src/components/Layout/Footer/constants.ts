export const XCN_MEDIUM_URL = 'https://medium.com/OnyxProtocol';
export const XCN_DISCORD_URL = 'https://discord.com/invite/onyxprotocol';
export const XCN_TELEGRAM_URL = 'https://t.me/OnyxOrg';
export const XCN_TWITTER_URL = 'https://twitter.com/OnyxProtocol';
export const XCN_GITHUB_URL = 'https://GitHub.com/Onyx-Protocol';
