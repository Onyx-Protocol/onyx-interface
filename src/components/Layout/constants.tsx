import Path from 'constants/path';

import { MenuItem } from './types';

/**
 * Updated by ozdav
 * Just commented to hide some menu links
 */
export const menuItems: MenuItem[] = [
  {
    href: Path.ROOT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.staking')
    i18nKey: 'layout.menuItems.stake',
    i18nTitleKey: 'layout.menuItems.stake',
    icon: 'stake',
  },
  // {
  //   href: Path.ROOT,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.dashboard')
  //   i18nKey: 'layout.menuItems.dashboard',
  //   i18nTitleKey: 'layout.menuItems.dashboard',
  //   icon: 'dashboard',
  // },
  // {
  //   href: Path.MARKETS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.markets')
  //   i18nKey: 'layout.menuItems.markets',
  //   i18nTitleKey: 'layout.menuItems.markets',
  //   icon: 'market',
  // },
  {
    href: Path.HISTORY,
    // Translation key: do not remove this comment
    // t('layout.menuItems.history')
    i18nKey: 'layout.menuItems.history',
    i18nTitleKey: 'layout.menuItems.history',
    icon: 'history',
  },
  {
    href: Path.SWAP,
    // Translation key: do not remove this comment
    // t('layout.menuItems.swap')
    i18nKey: 'layout.menuItems.swap',
    i18nTitleKey: 'layout.menuItems.swap',
    icon: 'convert',
    // isNew: true,
  },
  {
    href: Path.FARM,
    // Translation key: do not remove this comment
    // t('layout.menuItems.farm')
    i18nKey: 'layout.menuItems.farm',
    i18nTitleKey: 'layout.menuItems.farm',
    icon: 'farm',
    // isNew: true,
  },
  {
    href: Path.GOVERNANCE,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.governance',
    i18nTitleKey: 'layout.menuItems.governance',
    icon: 'vote',
  },
  {
    href: Path.AIAGENT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.aiagent',
    i18nTitleKey: 'layout.menuItems.aiagent',
    icon: 'aiagent',
    isNew: true,
  },
  // {
  //   href: Path.XCN,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.xcn',
  //   i18nTitleKey: 'layout.menuItems.xcn',
  //   icon: 'xcnOutlined',
  // },
  // {
  //   href: Path.LIQUIDATE,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.liquidate',
  //   i18nTitleKey: 'layout.menuItems.liquidate',
  //   icon: 'liquidate',
  // },
  // {
  //   href: Path.WPUNKS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.fpunk',
  //   i18nTitleKey: 'layout.menuItems.fpunk',
  //   icon: 'wpunksOutlined',
  // },
  // TO BE ACTIVATED ONCE CALCULATOR IS READY
  {
    href: Path.GOVERNANCE_LEADER_BOARD,
    // Translation key: do not remove this comment
    // t('layout.menuItems.xcn')
    i18nKey: 'layout.menuItems.leaderboard',
    i18nTitleKey: 'layout.menuItems.leaderboard',
    isNew: true,
    icon: 'xcnOutlined',
  },
  {
    href: Path.POINTS,
    // Translation key: do not remove this comment
    // t('layout.menuItems.points')
    i18nKey: 'layout.menuItems.points',
    i18nTitleKey: 'layout.menuItems.points',
    icon: 'gem',
    isNew: true,
    subMenu: [
      {
        href: Path.POINTS,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsDashboard')
        i18nKey: 'layout.menuItems.pointsDashboard',
        i18nTitleKey: 'layout.menuItems.pointsDashboard',
        icon: 'dashboard',
      },
      {
        href: Path.POINTS_LEADERBOARD,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsLeaderboard')
        i18nKey: 'layout.menuItems.pointsLeaderboard',
        i18nTitleKey: 'layout.menuItems.pointsLeaderboard',
        icon: 'person',
      },
    ],
  },
];
