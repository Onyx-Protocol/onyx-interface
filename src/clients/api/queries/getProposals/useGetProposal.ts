import { QueryObserverOptions, useQuery } from 'react-query';
import { Proposal } from 'types';

import { queryClient } from 'clients/api';
import getProposal from 'clients/api/queries/getProposals/getProposal';
import { GetProposalInput, GetProposalOutput } from 'clients/api/queries/getProposals/types';
import FunctionKey from 'constants/functionKey';

type Options = QueryObserverOptions<
  GetProposalOutput,
  Error,
  GetProposalOutput,
  GetProposalOutput,
  [FunctionKey.GET_PROPOSAL, GetProposalInput]
>;

const refetchStates = ['Pending', 'Active', 'Successded', 'Queued'];

// refetchInterval is set automatically with onSucess so it is excluded from being set manually
const useGetProposal = (params: GetProposalInput, options?: Omit<Options, 'refetchInterval'>) =>
  useQuery([FunctionKey.GET_PROPOSAL, params], () => getProposal(params), {
    onSuccess: (data: Proposal) => {
      const refetchInterval = refetchStates.includes(data.state) ? 5 * 60 * 1000 : 0;
      queryClient.setQueryDefaults([FunctionKey.GET_PROPOSAL, params], {
        refetchInterval,
      });
      if (options?.onSuccess) {
        options.onSuccess(data);
      }
    },
    ...options,
  });

export default useGetProposal;
