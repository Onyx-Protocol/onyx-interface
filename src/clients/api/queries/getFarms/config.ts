import { EthChainId } from 'types';

import { TOKENS, UNISWAP_TOKENS } from 'constants/tokens';

import { FarmConfig } from './types';

const farmConfig: FarmConfig[] = [
  {
    pid: 0,
    lpSymbol: 'XCN-WETH LP',
    lpAddresses: {
      [EthChainId.MAINNET]: '0x859f7092f56c43BB48bb46dE7119d9c799716CDF',
      [EthChainId.TESTNET]: '',
    },
    token: TOKENS.xcn,
    quoteToken: UNISWAP_TOKENS.weth,
  },

  // {
  //   pid: 2,
  //   lpSymbol: 'XCN-WETH LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0x006Df670C4735A18422B8c75F06d891D67e6B405',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: UNISWAP_TOKENS.weth,
  // },

  // {
  //   pid: 3,
  //   lpSymbol: 'XCN-USDT LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0xe346BcA6aDFCA5E6474a43E1d73270E43e80F41e',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: TOKENS.usdt,
  // },
];

export default farmConfig;
