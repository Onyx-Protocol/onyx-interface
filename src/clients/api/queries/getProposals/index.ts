import BigNumber from 'bignumber.js';
import config from 'config';
import { formatToProposal, getProposalsSubGraph } from 'utilities';

import { getGovernorBravoDelegateContract } from 'clients/contracts';
import getWeb3NoAccount from 'clients/web3/getWeb3NoAccount';

import { GetProposalsInput, GetProposalsOutput } from './types';

export * from './types';

const getProposals = async ({
  page = 0,
  limit = 5,
}: GetProposalsInput): Promise<GetProposalsOutput> => {
  const { proposals, count } = await getProposalsSubGraph(config.chainId, undefined, {
    offset: page * limit,
    limit,
  });

  const web3NoAccount = getWeb3NoAccount();
  const latestBlock = await web3NoAccount.eth.getBlock('latest');
  const governorBravoDelegateContract = getGovernorBravoDelegateContract(web3NoAccount);
  const quorumVotes = await governorBravoDelegateContract.methods.quorumVotes().call();

  const proposalsFormatted = await Promise.all(
    proposals.map(p =>
      formatToProposal(p, new BigNumber(quorumVotes), {
        latestBlockTimestamp: Number(latestBlock.timestamp),
        latestBlockNumber: latestBlock.number,
      }),
    ),
  );

  return { proposals: proposalsFormatted, total: count };
};

export default getProposals;
