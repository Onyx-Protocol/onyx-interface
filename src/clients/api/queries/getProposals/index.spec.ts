import config from 'config';
import { enableFetchMocks } from 'jest-fetch-mock';

import proposalResponse from '__mocks__/api/proposals.json';
import { SUBGRAPH_LINKS } from 'constants/endpoints';

import getProposals from '.';

enableFetchMocks();

describe('api/queries/getProposals', () => {
  test('returns formatted proposals', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        status: 200,
        data: {
          proposals: proposalResponse,
          proposalCounts: [{ count: proposalResponse.length }],
        },
      }),
    );

    const response = await getProposals({
      limit: 10,
      page: 2,
    });

    expect(fetchMock).toBeCalledWith(SUBGRAPH_LINKS[config.chainId].latest, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query proposalsQuery {
            proposals(
              first: 10,
              skip: 20,
              orderBy: createdBlockTimestamp,
              orderDirection: desc
              ${''}
            ) {
              id
              proposer
              targets
              values
              signatures
              callDatas
              startBlock
              endBlock
              description
              state
              eta
              forVotes
              againstVotes
              createdBlockNumber
              createdBlockTimestamp
              createdTransactionHash
              queuedBlockNumber
              queuedBlockTimestamp
              queuedTransactionHash
              executedBlockNumber
              executedBlockTimestamp
              executedTransactionHash
              canceledBlockNumber
              canceledBlockTimestamp
              canceledTransactionHash
            }
            
            proposalCounts(where: { id: "0" }) { count }
          }
        `,
      }),
    });

    expect(response).toMatchSnapshot();
  });

  test('Gets called with correct default arguments', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        status: 200,
        data: {
          proposals: proposalResponse,
          proposalCounts: [{ count: proposalResponse.length }],
        },
      }),
    );

    const response = await getProposals({});

    expect(fetchMock).toBeCalledWith(SUBGRAPH_LINKS[config.chainId].latest, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query proposalsQuery {
            proposals(
              first: 5,
              skip: 0,
              orderBy: createdBlockTimestamp,
              orderDirection: desc
              ${''}
            ) {
              id
              proposer
              targets
              values
              signatures
              callDatas
              startBlock
              endBlock
              description
              state
              eta
              forVotes
              againstVotes
              createdBlockNumber
              createdBlockTimestamp
              createdTransactionHash
              queuedBlockNumber
              queuedBlockTimestamp
              queuedTransactionHash
              executedBlockNumber
              executedBlockTimestamp
              executedTransactionHash
              canceledBlockNumber
              canceledBlockTimestamp
              canceledTransactionHash
            }
            
            proposalCounts(where: { id: "0" }) { count }
          }
        `,
      }),
    });

    expect(response).toMatchSnapshot();
  });
});
