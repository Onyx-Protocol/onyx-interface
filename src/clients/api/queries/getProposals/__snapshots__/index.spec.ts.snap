// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/getProposals Gets called with correct default arguments 1`] = `
Object {
  "proposals": Array [
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000002a300",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
        Object {
          "callData": "0x0000000000000000000000000000000000000000000000000000000000003200",
          "signature": "changeVotingPeriod(uint256)",
          "target": "0xdec2F31C3984F3440540DC78Ef21b1369d4eF767",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T20:52:40.000Z,
      "createdTxHash": "0x8b70679651c2e7cdeafbee647e2b52c5006a5c11d15f996edde95d020e45bda5",
      "description": Object {
        "description": " Chain Improvement Proposal 6 (CIP-6) will enable time lock parameters in the governance smart contracts to protect the protocol, the DAO, and its users. CIP-6 adds a 2 day voting window where users can place votes and a 2 day wait window before a proposal can be executed. Therefore, this 4 day window will protect the protocol from malicious attacks and set a reasonable time period for voting.",
        "title": " CIP-6 Secure Proposal Timelock",
        "version": "v1",
      },
      "endBlock": 4636946,
      "executedDate": 2022-04-22T21:12:53.000Z,
      "executedTxHash": "0xac2cead7d0a330a5bcfcdfba1ab5c39822d4b04052c5bfd4aaecd50b8c47506d",
      "forVotesWei": "2.324562998201e+26",
      "id": 6,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T21:01:30.000Z,
      "queuedTxHash": "0x53b9aaffd88bb56f4df1667fcb7058c484e90ead70f641166fee10658b1f5098",
      "startBlock": 4636916,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:57:45.000Z,
      "createdTxHash": "0xf36b8da226c217e95294aaee42f9646c460b46b317a610deedca8e8a15388930",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636708,
      "executedDate": undefined,
      "executedTxHash": undefined,
      "forVotesWei": "0",
      "id": 4,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": undefined,
      "queuedTxHash": undefined,
      "startBlock": 4636678,
      "state": "Executed",
      "totalVotesWei": "0",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x00000000000000000000000000000000000000000000000878678326eac90000",
          "signature": "setRewardPerblock(uint256)",
          "target": "0x23445c63FeEf8D85956dc0f19aDe87606D0e19A9",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T20:04:47.000Z,
      "createdTxHash": "0x3abef1f2811d39809f29a673f7c1cf0c4af6af3c479a03855fc82966d1d90319",
      "description": Object {
        "description": " Chain Improvement Proposal 5 (CIP-5) initiates the XCN Staking reward module to distribute XCN to stakers who are securing the Chain DAO and Chain Governance. The initial rewards will be set to 1 million XCN per day which is within the XCN allocated monthly per the latest Chain DAO Constitution. XCN Rewards will be distributed based on your staking weight in the protocol",
        "title": " CIP-5 Set Initial XCN Staking Rewards",
        "version": "v1",
      },
      "endBlock": 4636746,
      "executedDate": 2022-04-22T20:29:09.000Z,
      "executedTxHash": "0x1875dabc6bc38b7f716799179ec0c8eb1bc37c5b2cc88bbeaa097a2102077005",
      "forVotesWei": "2.324562998201e+26",
      "id": 5,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T20:11:11.000Z,
      "queuedTxHash": "0x985ec91cadcfb82e391faae2d7cb2fdaf4a0999d89aac009658a0f9aefb8c406",
      "startBlock": 4636716,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:47:04.000Z,
      "createdTxHash": "0x22d1f1e90bf5c37358514605b82a5c65f0b96fe1dde1e9e94bd879af319abfd2",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636666,
      "executedDate": undefined,
      "executedTxHash": undefined,
      "forVotesWei": "0",
      "id": 3,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": undefined,
      "queuedTxHash": undefined,
      "startBlock": 4636636,
      "state": "Executed",
      "totalVotesWei": "0",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:10:48.000Z,
      "createdTxHash": "0x3a748dab42685386e014189adecf71640bbdba37afa86c3e832b910f28fc0438",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636496,
      "executedDate": 2022-04-22T19:26:38.000Z,
      "executedTxHash": "0xd3b6d9ac10b004398bd5c384a9c3b24d2b597ee86beca5cee6a32924820e36bd",
      "forVotesWei": "2.324562998201e+26",
      "id": 2,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T19:19:14.000Z,
      "queuedTxHash": "0xf2fd3a9b442bf82bc7a3662a0d4d392821666ad86a81dbd30c1f1962a3994e91",
      "startBlock": 4636466,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
  ],
  "total": 5,
}
`;

exports[`api/queries/getProposals returns formatted proposals 1`] = `
Object {
  "proposals": Array [
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000002a300",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
        Object {
          "callData": "0x0000000000000000000000000000000000000000000000000000000000003200",
          "signature": "changeVotingPeriod(uint256)",
          "target": "0xdec2F31C3984F3440540DC78Ef21b1369d4eF767",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T20:52:40.000Z,
      "createdTxHash": "0x8b70679651c2e7cdeafbee647e2b52c5006a5c11d15f996edde95d020e45bda5",
      "description": Object {
        "description": " Chain Improvement Proposal 6 (CIP-6) will enable time lock parameters in the governance smart contracts to protect the protocol, the DAO, and its users. CIP-6 adds a 2 day voting window where users can place votes and a 2 day wait window before a proposal can be executed. Therefore, this 4 day window will protect the protocol from malicious attacks and set a reasonable time period for voting.",
        "title": " CIP-6 Secure Proposal Timelock",
        "version": "v1",
      },
      "endBlock": 4636946,
      "executedDate": 2022-04-22T21:12:53.000Z,
      "executedTxHash": "0xac2cead7d0a330a5bcfcdfba1ab5c39822d4b04052c5bfd4aaecd50b8c47506d",
      "forVotesWei": "2.324562998201e+26",
      "id": 6,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T21:01:30.000Z,
      "queuedTxHash": "0x53b9aaffd88bb56f4df1667fcb7058c484e90ead70f641166fee10658b1f5098",
      "startBlock": 4636916,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:57:45.000Z,
      "createdTxHash": "0xf36b8da226c217e95294aaee42f9646c460b46b317a610deedca8e8a15388930",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636708,
      "executedDate": undefined,
      "executedTxHash": undefined,
      "forVotesWei": "0",
      "id": 4,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": undefined,
      "queuedTxHash": undefined,
      "startBlock": 4636678,
      "state": "Executed",
      "totalVotesWei": "0",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x00000000000000000000000000000000000000000000000878678326eac90000",
          "signature": "setRewardPerblock(uint256)",
          "target": "0x23445c63FeEf8D85956dc0f19aDe87606D0e19A9",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T20:04:47.000Z,
      "createdTxHash": "0x3abef1f2811d39809f29a673f7c1cf0c4af6af3c479a03855fc82966d1d90319",
      "description": Object {
        "description": " Chain Improvement Proposal 5 (CIP-5) initiates the XCN Staking reward module to distribute XCN to stakers who are securing the Chain DAO and Chain Governance. The initial rewards will be set to 1 million XCN per day which is within the XCN allocated monthly per the latest Chain DAO Constitution. XCN Rewards will be distributed based on your staking weight in the protocol",
        "title": " CIP-5 Set Initial XCN Staking Rewards",
        "version": "v1",
      },
      "endBlock": 4636746,
      "executedDate": 2022-04-22T20:29:09.000Z,
      "executedTxHash": "0x1875dabc6bc38b7f716799179ec0c8eb1bc37c5b2cc88bbeaa097a2102077005",
      "forVotesWei": "2.324562998201e+26",
      "id": 5,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T20:11:11.000Z,
      "queuedTxHash": "0x985ec91cadcfb82e391faae2d7cb2fdaf4a0999d89aac009658a0f9aefb8c406",
      "startBlock": 4636716,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:47:04.000Z,
      "createdTxHash": "0x22d1f1e90bf5c37358514605b82a5c65f0b96fe1dde1e9e94bd879af319abfd2",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636666,
      "executedDate": undefined,
      "executedTxHash": undefined,
      "forVotesWei": "0",
      "id": 3,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": undefined,
      "queuedTxHash": undefined,
      "startBlock": 4636636,
      "state": "Executed",
      "totalVotesWei": "0",
    },
    Object {
      "actions": Array [
        Object {
          "callData": "0x000000000000000000000000000000000000000000000000000000000000012c",
          "signature": "setDelay(uint256)",
          "target": "0x08eDF0F2AF8672029eb445742B3b4072c6158DF3",
          "value": "0",
        },
      ],
      "againstVotesWei": "0",
      "cancelDate": undefined,
      "cancelTxHash": undefined,
      "createdDate": 2022-04-22T19:10:48.000Z,
      "createdTxHash": "0x3a748dab42685386e014189adecf71640bbdba37afa86c3e832b910f28fc0438",
      "description": Object {
        "description": " Chain Improvement Proposal 4 (CIP-4) enables the Chain Governance module to update the Timelock smart contract for final mainnet testing purposes.",
        "title": " CIP-4 Pre-Governance Activation",
        "version": "v1",
      },
      "endBlock": 4636496,
      "executedDate": 2022-04-22T19:26:38.000Z,
      "executedTxHash": "0xd3b6d9ac10b004398bd5c384a9c3b24d2b597ee86beca5cee6a32924820e36bd",
      "forVotesWei": "2.324562998201e+26",
      "id": 2,
      "isEnded": true,
      "isStarted": true,
      "proposer": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "queuedDate": 2022-04-22T19:19:14.000Z,
      "queuedTxHash": "0xf2fd3a9b442bf82bc7a3662a0d4d392821666ad86a81dbd30c1f1962a3994e91",
      "startBlock": 4636466,
      "state": "Executed",
      "totalVotesWei": "2.324562998201e+26",
    },
  ],
  "total": 5,
}
`;
